components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java

```diff
@@ -120,6 +128,7 @@ public AuthenticationContext authenticate(String jwtToken, MessageContext synCtx
                 throw new APISecurityException(APISecurityConstants.API_AUTH_INVALID_CREDENTIALS,
                         "Invalid JWT token. Failed to decode the token.", e);
             }
+            log.debug("Verifying signature of JWT");
```

Comment:
Do we know which token has this issue? Otherwise priting this log won't be much useful.